app/controllers/provider_orders_controller.rb

```diff
@@ -54,49 +54,54 @@ def resubmit
+      order['closed_date'] = provider_order['ClosedDate']
 
-    provider_order_guid = provider_order['Guid']
-    name_guids = echo_client.get_order_item_names_by_provider_order(echo_provider_token, provider_order_guid).parsed_body.fetch('Item', {})
+      provider_order_guid = provider_order['Guid']
+      name_guids = echo_client.get_order_item_names_by_provider_order(echo_provider_token, provider_order_guid).parsed_body.fetch('Item', {})
 
-    item_guids = Array.wrap(name_guids).map { |name_guid| name_guid.fetch('Guid', '') }
+      item_guids = Array.wrap(name_guids).map { |name_guid| name_guid.fetch('Guid', '') }
 
-    items = echo_client.get_order_items(echo_provider_token, item_guids).parsed_body.fetch('Item', {})
+      items = echo_client.get_order_items(echo_provider_token, item_guids).parsed_body.fetch('Item', {})
 
-    catalog_items = []
-    Array.wrap(items).each do |order_item|
-      order_item_detail = order_item.fetch('OrderItemDetail', {})
+      catalog_items = []
+      Array.wrap(items).each do |order_item|
+        order_item_detail = order_item.fetch('OrderItemDetail', {})
 
-      catalog_item = {}
-      catalog_item['item_guid'] = order_item.fetch('ItemGuid', '')
-      catalog_item['status'] = order_item_detail.fetch('State', '')
-      catalog_item['data_granule_id'] = order_item_detail.fetch('GranuleUR', '')
-      catalog_item['local_granule_id'] = order_item_detail.fetch('ProducerGranuleId', '')
-      option_selection = order_item.fetch('OptionSelection', {})
-      catalog_item['option_selection'] = {}
-      catalog_item['option_selection']['name'] = option_selection.fetch('Name', '')
-      catalog_item['option_selection']['content'] = option_selection.fetch('Content', '')
+        catalog_item = {}
+        catalog_item['item_guid'] = order_item.fetch('ItemGuid', '')
+        catalog_item['status'] = order_item_detail.fetch('State', '')
+        catalog_item['data_granule_id'] = order_item_detail.fetch('GranuleUR', '')
+        catalog_item['local_granule_id'] = order_item_detail.fetch('ProducerGranuleId', '')
+        option_selection = order_item.fetch('OptionSelection', {})
+        catalog_item['option_selection'] = {}
+        catalog_item['option_selection']['name'] = option_selection.fetch('Name', '')
+        catalog_item['option_selection']['content'] = option_selection.fetch('Content', '')
 
-      catalog_items << catalog_item
-    end
+        catalog_items << catalog_item
+      end
 
-    order['catalog_items'] = catalog_items.sort { |a, b| a['item_guid'] <=> b['item_guid'] }
-    order['status_messages'] = provider_order.fetch('StatusMessage', '').split("\n").map { |m| m.split(' : ') }
+      order['catalog_items'] = catalog_items.sort_by { |a| a['item_guid'] }
+      order['status_messages'] = provider_order.fetch('StatusMessage', '').split("\n").map { |m| m.split(' : ') }
 
-    order
+      order
+    else
+      { 'error' => order_response.parsed_body.fetch('faultstring', true) }
```

Comment:
1. in `lib/echo/response.rb`, there are a few different methods to get at the response body and errors. that’s where I found `parsed_body`, but it seems there are more detailed/helpful methods for `error_message` and `error_code` among others. We should definitely switch how we are grabbing the error message to use `error_message`, and if it is helpful we may want to use some of the other methods also. we may also want to modify `error_message` because it does not provide a default value to `fetch` which can produce an exception.
2. since the error message is something we are providing to the user to contact OPS with, we should probably have something other than `true`. I had initially thought we should use something similar to our I18n default error flash messages (in `config/locales/en.yml`), but this is not a flash message and is more similar to what we do in CollectionsController#set_collection_error_data, so I think the setup is fine, but we should change the default message to be something more informative than just `true` for this error so OPS has more information to discuss with us in the case where the response does not provide an error message in the way we are expecting.